ui/apps/platform/src/Containers/Vulnerabilities/VulnerablityReporting/api/useWatchLastSnapshotForReports.ts

```diff
@@ -36,64 +27,57 @@ async function fetchLastReportJobForConfiguration(
     return reportSnapshot[0] ?? null;
 }
 
-const defaultResult = {
-    reportSnapshots: {},
-    isLoading: false,
-    error: null,
+type ReportSnapshotLookup = Record<string, ReportSnapshot | null>;
+
+type Result = {
+    reportSnapshots: ReportSnapshotLookup;
+    isLoading: boolean;
+    error: string | null;
+};
+
+export type FetchLastSnapshotReturn = Result & {
+    fetchSnapshots: () => void;
 };
 
 export function useWatchLastSnapshotForReports(
     reportConfigurations: ReportConfiguration | ReportConfiguration[] | null
 ): FetchLastSnapshotReturn {
-    const [result, setResult] = useState<Result>(defaultResult);
-
-    const fetchSnapshots = useCallback(async () => {
+    const fetchSnapshotsCallback = useCallback(async () => {
         if (!reportConfigurations) {
-            return;
+            const result: ReportSnapshotLookup = {};
+            return Promise.resolve(result);
         }
 
-        setResult((prevResult) => ({
-            ...prevResult,
-            isLoading: true,
-            error: null,
-        }));
+        const promise: Promise<ReportSnapshotLookup> = new Promise((resolve, reject) => {
+            const configurations = Array.isArray(reportConfigurations)
```

Comment:
A question, not necessarily a change request: what are pro and con of `reportConfigurations: ReportConfiguration[]` as argument and caller is responsible for single (and possibly initially `null` case:

```js
useWatchLastSnapshotForReports(reportConfiguration ? [reportConfiguration] : [])
```

Reply:
I tried this approach before but ran into some strange behavior. Each time the component re-renders, a new object gets passed to the hook, which can cause unintended side effects. We could handle this inside the component itself using `useMemo`, but that would add complexity. It might be better to manage this logic within the hook itself to keep it simple and easier to use.

Reply:
Thank you, that makes sense!